app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/LocalFileTransferActivity.java

```diff
@@ -0,0 +1,437 @@
+    }
+
+    try {
+      network_enabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+    } catch (Exception ex) {
+      ex.printStackTrace();
+    }
+
+    return (gps_enabled || network_enabled);
+  }
+
+  private void requestEnableLocationServices() {
+    alertDialogShower.show(KiwixDialog.EnableLocationServices.INSTANCE,
+        new Function0<Unit>() {
+          @Override public Unit invoke() {
+            startActivityForResult(new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS),
+                REQUEST_ENABLE_LOCATION_SERVICES);
+            return Unit.INSTANCE;
+          }
+        },
+        new Function0<Unit>() {
+          @Override public Unit invoke() {
+            showToast(LocalFileTransferActivity.this, R.string.discovery_needs_location,
+                Toast.LENGTH_SHORT);
+            return Unit.INSTANCE;
+          }
+        });
+  }
+
+  private void requestEnableWifiP2pServices() {
+    alertDialogShower.show(KiwixDialog.EnableWifiP2pServices.INSTANCE,
+        new Function0<Unit>() {
+          @Override public Unit invoke() {
+            startActivity(new Intent(Settings.ACTION_WIFI_SETTINGS));
+            return Unit.INSTANCE;
+          }
+        },
+        new Function0<Unit>() {
+          @Override public Unit invoke() {
+            showToast(LocalFileTransferActivity.this, R.string.discovery_needs_wifi,
+                Toast.LENGTH_SHORT);
+            return Unit.INSTANCE;
+          }
+        });
+  }
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+    switch (requestCode) {
+      case REQUEST_ENABLE_LOCATION_SERVICES: {
+        LocationManager locationManager =
+            (LocationManager) getApplicationContext().getSystemService(Context.LOCATION_SERVICE);
+
+        if (!(locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
+            || locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER))) {
+          // If neither provider is enabled
+          showToast(this, R.string.permission_refused_location, Toast.LENGTH_LONG);
+        }
+        break;
+      }
+
+      default: {
+        super.onActivityResult(requestCode, resultCode, data);
+        break;
+      }
+    }
+  }
+
+  /* Miscellaneous helper methods */
+  static void showToast(Context context, int stringResource, int duration) {
+    showToast(context, context.getString(stringResource), duration);
+  }
+
+  static void showToast(Context context, String text, int duration) {
+    Toast.makeText(context, text, duration).show();
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    wifiDirectManager.registerWifiDirectBroadcastRecevier();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    wifiDirectManager.unregisterWifiDirectBroadcastRecevier();
+  }
+
+  @Override protected void onDestroy() {
+    super.onDestroy();
+    wifiDirectManager.destroyWifiDirectManager();
```

Comment:
I would call this `stop`, also there is [this](http://stackoverflow.com/questions/18821481/what-is-the-correct-order-of-calling-superclass-methods-in-onpause-onstop-and-o) irritating SO quetion you should read, tl;dr
```
Methods you override that are part of component creation (onCreate(), onStart(), onResume(), etc.), you should chain to the superclass as the first statement, to ensure that Android has its chance to do its work before you attempt to do something that relies upon that work having been done. 
Methods you override that are part of component destruction (onPause(), onStop(), onDestroy(), etc.), you should do your work first and chain to the superclass as the last thing. That way, in case Android cleans up something that your work depends upon, you will have done your work first. 
```
but only sometimes

Reply:
Changed the name and order of calls - [commit](https://github.com/kiwix/kiwix-android/pull/1276/commits/32bfe70643db1b647479719fdc97595eaaa5ae21#diff-e1d59ae6e15dcb502e125fb6e243d15dR393)
I don't think I would have ever thought about this until I was actually affected by it. Even then pinpointing the cause w/out prior knowledge would have taken time. Thanks!